axelrod/match_generator.py

```diff
@@ -207,3 +208,42 @@ def build_match_chunks(self):
 
     def __len__(self):
         return len(self.edges)
+
+
+class ProbEndSpatialMatches(SpatialMatches, ProbEndRoundRobinMatches):
+    """
+    A class that generates spatially-structured prob ending matches.
+    In these matches, players interact only with their neighbors rather than the
+    entire population. This reduces to a well-mixed population when the spatial
+    graph is a complete graph.
+
+    Parameters
+    ----------
+    players : list
+        A list of axelrod.Player objects
+    prob_end : float
+        The probability that a turn of a Match is the last
+    game : axelrod.Game
+        The game object used to score the match
+    repetitions : int
+        The number of repetitions of a given match
+    edges : list
+        A list of tuples containing the existing edges
+    """
+
+    def __init__(self, players, prob_end, game, repetitions, noise, edges):
+
+        player_indices = list(range(len(players)))
+        node_indices = sorted(set([node for edge in edges for node in edge]))
+        if player_indices != node_indices:
```

Comment:
If I understand correctly then these lines are just a check and the results are not needed. It might be more efficient to use sets and check for membership rather than build two lists, sort one, and check if they are equal. I think the current solution is `O(n^2)` and my proposal is better.